src/main/java/Duke.java

```diff
@@ -1,10 +1,179 @@
+    }
+    public static String parseInput(String[] inputs){
+        String todoInput = "";
+        for (int i = 1; i < inputs.length; i ++) {
+            todoInput += inputs[i];
+            todoInput += " ";
+        }
+        return todoInput;
+    }
+    public static void inputTodo(String[] inputs, int taskCounter) {
+        String todoInput = parseInput(inputs);
+        Todo todo = new Todo(todoInput);
+        tasks.add(todo);
+        taskCounter += 1;
+        printAddTodo(todo, taskCounter);
+    }
+    private static boolean readInput() throws EmptyInputException, IllegalInputException {
+        Scanner sc = new Scanner(System.in);
+        String input = "";
+        input = sc.nextLine();
+
+        int taskCounter = 0;
+
+
+        while (!input.equals("bye")) {
+
+
+            String[] inputs = input.split(" ");
+            String command  = inputs[0];
+
+            printLine();
+
+            switch (command) {
+            case "list":
+                printList(taskCounter);
+                break;
+            case "mark":
+                int taskIndex = Integer.parseInt(inputs[1]);
+                if (taskIndex > tasks.size() || taskIndex <= 0) throw  new IllegalInputException();
+                tasks.get(taskIndex-1).markAsDone();
+                System.out.println("I have marked this task as done");
+                printList(taskCounter);
+                break;
+            case "unmark":
+                taskIndex = Integer.parseInt(inputs[1]);
+                if (taskIndex > tasks.size() || taskIndex <= 0) throw  new IllegalInputException();
+                tasks.get(taskIndex-1).markAsNotDone();
+                System.out.println("I have unmarked this task");
+                printList(taskCounter);
+                break;
+            case "todo":
+                inputTodo(inputs, taskCounter);
+                break;
+            case "deadline":
+                String deadlineInput = "";
+                String deadlineDeadline = "";
+                boolean isDeadline = false;
+                for (int i = 1; i < inputs.length; i ++) {
+                    if (inputs[i].contains("/")) {
+                        isDeadline = true;
+                        continue;
+                    }
+                    if (isDeadline) {
+                        deadlineDeadline += inputs[i];
+                        deadlineDeadline += " ";
+                    }
+                    else {
+                        deadlineInput += inputs[i];
+                        deadlineInput += " ";
+                    }
+                }
+                if (!isDeadline) throw new IllegalInputException();
+                Deadline deadline = new Deadline(deadlineInput, deadlineDeadline);
+                tasks.add(deadline); taskCounter += 1;
+                printAddDeadline(deadline, taskCounter);
+                break;
+            case "event":
+                String eventInput = "";
```

Comment:
Code quality: Avoid arrowhead code

Again, can be resolved by breaking down into smaller methods to enhance readability.